src/drivers/zotac_zone/driver.rs

```diff
@@ -0,0 +1,70 @@
+use std::error::Error;
+
+use udev::Device;
+
+use crate::udev::device::{AttributeGetter, AttributeSetter, UdevDevice};
+
+// Hardware ID's
+const ZONE_PID: u16 = 0x1590;
+pub const PIDS: [u16; 1] = [ZONE_PID];
+pub const VID: u16 = 0x1ee9;
+pub const VID_ALT: u16 = 0x1e19;
+pub const VIDS: [u16; 2] = [VID, VID_ALT];
+
+pub struct Driver {
+    _device: UdevDevice,
+}
+
+impl Driver {
+    pub fn new(udevice: UdevDevice) -> Result<Self, Box<dyn Error + Send + Sync>> {
+        let vid = udevice.id_vendor();
+        let pid = udevice.id_product();
+        if !VIDS.contains(&vid) || !PIDS.contains(&pid) {
+            return Err(format!("'{}' is not an Zotac Zone controller", udevice.devnode()).into());
+        }
+
+        // Set the controller buttons to the correct values at startup
+        let mut device = udevice.get_device()?;
+        let Some(parent) = device.parent() else {
+            return Err("Failed to get device parent".into());
+        };
+        let Some(driver) = parent.driver() else {
+            return Err("Failed to identify device driver".into());
+        };
+        // Apply settings for the controller
+        let drv_name = driver.to_str().unwrap();
+        if drv_name == "zotac_zone_hid" || drv_name == "hid_zotac_zone" {
+            let if_num = device.get_attribute_from_tree("bInterfaceNumber");
```

Comment:
This can be simplified and done as part of the driver discovery now.

We're getting the driver interface ID in the source/hidraw.rs
https://github.com/ShadowBlip/InputPlumber/blob/561df7bff0355c8efa4e3b2ce4cfed3c5e3a31e6/src/input/source/hidraw.rs#L384

This was used to load the correct driver for each of the legos interfaces.

https://github.com/ShadowBlip/InputPlumber/blob/main/src/input/source/hidraw.rs#L427-L452

If you set a `pub const CFG_IID: i32 = 0x3;` at the top of the file you can check for it the same way there.

Reply:
Simplified the whole block